uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-fs/src/main/java/org/uberfire/java/nio/fs/file/SimpleFileSystemProvider.java

```diff
@@ -252,117 +236,41 @@ private boolean shouldCreateOrOpenAByteChannel(final Set<? extends OpenOption> o
         }
+    
         final File[] content = file.listFiles();
-
         if (content == null) {
             throw new NotDirectoryException(dir.toString());
         }
-
-        return new DirectoryStream<Path>() {
-            boolean isClosed = false;
-
-            @Override
-            public void close() throws IOException {
-                if (isClosed) {
-                    throw new IOException("This stream is closed.");
-                }
-                isClosed = true;
-            }
-
-            @Override
-            public Iterator<Path> iterator() {
-                if (isClosed) {
-                    throw new IOException("This stream is closed.");
-                }
-                return new Iterator<Path>() {
-                    public boolean atEof = false;
-                    private int i = -1;
-                    private Path nextEntry = null;
-
-                    @Override
-                    public boolean hasNext() {
-                        if (nextEntry == null && !atEof) {
-                            nextEntry = readNextEntry();
-                        }
-                        return nextEntry != null;
-                    }
-
-                    @Override
-                    public Path next() {
-                        final Path result;
-                        if (nextEntry == null && !atEof) {
-                            result = readNextEntry();
-                        } else {
-                            result = nextEntry;
-                            nextEntry = null;
-                        }
-                        if (result == null) {
-                            throw new NoSuchElementException();
-                        }
-                        return result;
-                    }
-
-                    private Path readNextEntry() {
-                        if (atEof) {
-                            return null;
-                        }
-
-                        Path result = null;
-                        while (true) {
-                            i++;
-                            if (i >= content.length) {
-                                atEof = true;
-                                break;
-                            }
-
-                            final Path path = GeneralPathImpl.newFromFile(getDefaultFileSystem(),
-                                                                          content[i]);
-                            if (filter.accept(path)) {
-                                result = path;
-                                break;
-                            }
-                        }
-
-                        return result;
-                    }
-
-                    @Override
-                    public void remove() {
-                        throw new UnsupportedOperationException();
-                    }
-                };
-            }
-        };
+    
+        return Arrays.stream(content)
```

Comment:
i may be wrong, but the original seemed more efficient.

Reply:
Actually, you are right (Sharp eye!). Comparing to original implementation, the extra step is to convert from an File[] to Path[]. The reason behind it is that the base implementation of 'newDirectoryStream' is 90% reusable for K8SFileSystemProvider except those few lines of code with dependency on java.io.File. (https://github.com/kiegroup/appformer/blob/eec892e30e1d1570e22ead844385a31ea78b4b61/uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-fs/src/main/java/org/uberfire/java/nio/fs/file/SimpleFileSystemProvider.java#L272-L282), which have been refactored out into a method with the following signature:
`protected Path[] getDirectoryContent(final Path dir)`
So that K8SFileSystemProvider can provide an alternative implementation.

Reply:
I am wondering if K8SFileSystemProvider could do composition instead of inheritance. WDYT @rhtevan ?